### PR TITLE
Manage child pipelines

### PR DESCRIPTION
**Pull request by sagar1312**
## Context
Changes were made in https://github.com/screwdriver-cd/models/pull/534 and https://github.com/screwdriver-cd/screwdriver/pull/2724 to stop auto deletion of child pipelines when the SCM URLs removed/modified in the external config.

Child pipeline state will be set to `INACTIVE` when the associated SCM URL has been removed in the external configuration.
Pipeline admin can review and take necessary action on such inactive child pipeline

- reactivate the pipeline by restoring the SCM URL in the external config
- explicitly delete the pipeline from the UI/API

## Objective
This PR includes changes in the UI to notify the user if inactive child pipeline exists and provide an option to delete it.
Here is the page level breakdown of UI changes...

### Child Pipelines list view
1. Display info message at the top when there is one or more inactive child pipeline.
2. Add a new column `STATE` in the table
3. Add a new column `ACTION` which will include trash icon for inactive child pipeline row, which when clicked will delete that pipeline
4. Disable `Start All` button if all the child pipelines are inactive

image

### Pipeline Events view
When the pipeline is inactive
1. Display info message at the top to indicate the same
2. Disable `Start` button to prevent creation of new event
3. Remove `Start/Restart pipeline from here` from the workflow graph node menu to prevent creation of event

image

### Build Details view
When the pipeline is inactive
1. Disable 'Restart' button to prevent restarting the build

image

## References

https://github.com/screwdriver-cd/screwdriver/issues/2580

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by tkyi — DISMISSED** [review]

File: app/pipeline/child-pipelines/template.hbs [hunk 1]
```diff
@@ -2,6 +2,10 @@
   {{pipeline-nav pipeline=pipeline}}
 {{/if}}
 
-{{pipeline-list pipelines=pipelines pipeline=pipeline}}
+{{#if hasInactivePipelines}}
+  {{info-message message="You have one or more inactive pipelines. You can activate these by adding back the corresponding scm URL in the yaml configuration." type="info" icon="exclamation-triangle"}}
```
Comment (jithine) [inline app/pipeline/child-pipelines/template.hbs:6]: scm to be in caps

File: app/pipeline/events/template.hbs [hunk 2]
```diff
@@ -9,6 +9,10 @@
 
 {{info-message message=errorMessage type="warning" icon="exclamation-triangle" scmContext=pipeline.scmContext}}
 
+{{#if isInactivePipeline}}
+  {{info-message message="Pipeline is inactive and no new events can be started. You can activate the pipeline by adding the scm URL in the parent pipeline yaml configuration." type="info" icon="exclamation-triangle"}}
```
Comment (jithine) [inline app/pipeline/events/template.hbs:13]: scm in caps

**Review by jithine — DISMISSED** [review]

File: app/pipeline/child-pipelines/template.hbs (hunk 1 above)
Comment (sagar1312) [reply app/pipeline/child-pipelines/template.hbs]: Done

File: app/pipeline/events/template.hbs (hunk 2 above)
Comment (sagar1312) [reply app/pipeline/events/template.hbs]: Done

**Pull request merged**